### ISSUE TITLE
SOAP Scanner add-on leads to errors with Java 9+

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#4037]
Split from #2602 (https://github.com/zaproxy/zaproxy/issues/2602#issuecomment-334855554).

The SOAP Scanner is failing with:
```
15374 [ZAP-BootstrapGUI] ERROR org.parosproxy.paros.extension.ExtensionLoader  - javax/xml/soap/SOAPException
java.lang.NoClassDefFoundError: javax/xml/soap/SOAPException
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3110)
	at java.base/java.lang.Class.getConstructor0(Class.java:3315)
	at java.base/java.lang.Class.getConstructor(Class.java:2108)
	at org.zaproxy.zap.control.AddOnLoaderUtils.loadAndInstantiateClassImpl(AddOnLoaderUtils.java:111)
	at org.zaproxy.zap.control.AddOnLoaderUtils.loadDeclaredClasses(AddOnLoaderUtils.java:151)
	at org.zaproxy.zap.control.AddOnLoaderUtils.getActiveScanRules(AddOnLoaderUtils.java:177)
	at org.zaproxy.zap.control.AddOnLoader.getActiveScanRules(AddOnLoader.java:791)
	at org.parosproxy.paros.core.scanner.PluginFactory.initPlugins(PluginFactory.java:99)
	at org.parosproxy.paros.core.scanner.PluginFactory.getLoadedPlugins(PluginFactory.java:131)
	at org.parosproxy.paros.core.scanner.PluginFactory.loadAllPlugin(PluginFactory.java:397)
	at org.zaproxy.zap.extension.ascan.ScanPolicy.<init>(ScanPolicy.java:22)
	at org.zaproxy.zap.extension.ascan.PolicyManager.getAllPolicyNames(PolicyManager.java:68)
	at org.zaproxy.zap.extension.ascan.PolicyManager.init(PolicyManager.java:51)
	at org.zaproxy.zap.extension.ascan.ExtensionActiveScan.postInit(ExtensionActiveScan.java:130)
	at org.parosproxy.paros.extension.ExtensionLoader.hookAllExtension(ExtensionLoader.java:793)
	at org.parosproxy.paros.extension.ExtensionLoader.startLifeCycle(ExtensionLoader.java:664)
	at org.parosproxy.paros.control.AbstractControl.loadExtension(AbstractControl.java:57)
	at org.parosproxy.paros.control.Control.init(Control.java:123)
	at org.parosproxy.paros.control.Control.initSingletonWithView(Control.java:299)
	at org.zaproxy.zap.GuiBootstrap.initControlAndPostViewInit(GuiBootstrap.java:258)
	at org.zaproxy.zap.GuiBootstrap.access$100(GuiBootstrap.java:70)
	at org.zaproxy.zap.GuiBootstrap$2.run(GuiBootstrap.java:200)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.lang.ClassNotFoundException
	at org.zaproxy.zap.control.AddOnClassLoader.findClass(AddOnClassLoader.java:256)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:563)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:496)
	... 24 more
```

Add-on:
SOAP Scanner, version 3.

**Comment by thc202** [comment]
This will be addressed during the migration to Gradle (to avoid pushing more binaries to the repo), by depending on (and thus bundling) the required library.

**Comment by thc202** [comment]
Done in zaproxy/zap-extensions#1974.

**Comment by thc202** [comment]
fwiw, while the above exception was fixed the add-on was not working correctly, the classes were not being properly loaded... That should be fixed in upcoming changes as part of #4866.
